aws/resource_aws_api_gateway_integration.go

```diff
@@ -131,14 +146,26 @@ func resourceAwsApiGatewayIntegrationCreate(d *schema.ResourceData, meta interfa
 	conn := meta.(*AWSClient).apigateway
 
 	log.Print("[DEBUG] Creating API Gateway Integration")
+
+	connectionType := aws.String(d.Get("connection_type").(string))
+	var connectionId *string
+	if *connectionType == apigateway.ConnectionTypeVpcLink {
+		if _, ok := d.GetOk("connection_id"); !ok {
+			return fmt.Errorf("connection_id required when connection_type set to VPC_LINK")
```

Comment:
We can move this to a `CustomizeDiff` for plan time validation at some point 👍

Reply:
I tried but didn't manage to do it because of question https://github.com/hashicorp/terraform/issues/17595